spec/unit/type/exec_spec.rb

```diff
@@ -408,16 +421,40 @@ def test(command, valid)
         expect(@test[param]).to eq(input)
       end
     end
+
+    describe "when given an array of arrays as input" do
+      let(:test) { Puppet::Type.type(:exec).new(:name => @executable) }
+
+      it "accepts the array when all commands return valid" do
+        input = [['one, -a'], ['two', '-b'], 'three']
+        expect(test.provider).to receive(:validatecmd).exactly(input.length).times.and_return(true)
+        test[param] = input
+        expect(test[param]).to eq(input)
```

Comment:
Maybe I do not understand ruby well enough, but this assertion seems strange to me, as it only exactly validates what was assigned one line above. And I don't really see where you have the actual expected output (accepts vs rejects the array)

Reply:
It's a bit weird because the `Puppet::Type#[]=` method is overriden here https://github.com/puppetlabs/puppet/blob/a652a22156e6637c0cad4e0d3aa980a2398d0801/lib/puppet/type.rb#L655

So setting a property/parameter like this also triggers the validation of the parameter

Reply:
Thanks for the explanation, so if this line checks that the parameter is deemed valid, how do the test cases below (Line 441, Line 448) specify that the parameter is not valid? Shouldn't there be an error raised in that case that the test case then expects?

Reply:
Previous tests did not make any tests, updated with new cases that raise error when the command is invalid.
The shared examples are run for `:command, :onlyif, :unless, :refresh`